apps/studio/src/server/modules/resource/resource.service.ts

```diff
@@ -0,0 +1,48 @@
+import { db } from '../database'
+
+export const getPages = () => {
+  return (
+    db
+      .selectFrom('Resource')
+      // TODO: write a test to verify this query behaviour
+      .where('blobId', '!=', null)
+      .selectAll()
+      .execute()
+  )
+}
+
+export const getFolders = () =>
+  db
+    .selectFrom('Resource')
+    // TODO: write a test to verify this query behaviour
+    .where('blobId', '=', null)
+    .selectAll()
+    .execute()
+
+// NOTE: Base method for retrieving a resource - no distinction made on whether `blobId` exists
+const getById = (id: string) => db.selectFrom('Resource').where('id', '=', id)
+
+// NOTE: Throw here to fail early if our invariant that a page has a `blobId` is violated
+export const getFullPageById = (id: string) => {
+  return getById(id)
+    .where('blobId', '!=', null)
+    .innerJoin('Blob', 'blobId', 'Blob.id')
+    .selectAll()
+    .executeTakeFirstOrThrow()
+}
+
+export const getPageById = (id: string) => {
+  return getById(id)
+    .where('blobId', '!=', null)
+    .selectAll()
+    .executeTakeFirstOrThrow()
+}
```

Comment:
Hmm when would we need to make use of this endpoint? If we're looking for a quick view this would be retrieved from getPages, and the full view is covered by `getFullPageById`

Reply:
this isn't an endpoint - was thinking if we wanted to just return the meta of teh page without content, such as in a folder view